ambry-frontend/src/main/java/com/github/ambry/frontend/NamedBlobPutHandler.java

```diff
@@ -488,6 +490,34 @@ private void verifyChunkAccountAndContainer(String chunkBlobId, BlobProperties s
       }
     }
 
+    /**
+     * Verify that the reserved metadata id for the specified chunkId is same as seen for previous chunks.
+     * Also return the chunk's reserved metadata id.
+     * @param metadata {@link Map} of metadata set in the signed ids.
+     * @param reservedMetadataBlobId Reserved metadata id for the chunks. Can be {@code null}.
+     * @param chunkId The chunk id.
+     * @return The reserved metadata id.
+     * @throws RestServiceException in case of any exception.
+     */
+    private String getAndVerifyReservedMetadataBlobId(Map<String, String> metadata, String reservedMetadataBlobId,
+        String chunkId) throws RestServiceException {
+      String chunkReservedMetadataBlobId = RestUtils.getHeader(metadata, RestUtils.Headers.RESERVED_METADATA_ID, false);
+      if (reservedMetadataBlobId == null && chunkReservedMetadataBlobId == null) {
+        ReservedMetadataIdMetrics.getReservedMetadataIdMetrics(
```

Comment:
This method will be called only for stitch blob requests. So even if someone is stitching only one data chunk, a metadata chunk will still be created. Also, when someone created the POST-signed-URL to upload the data chunk, then the metadata chunk should have been reserved and that metadata chunk should ultimately be used for saving the stitched blob.

When no metadata chunk is reserved in the signed-id of the chunk to stitch, then this will increment the metric. This is to give us some metric to track that there are some stitch blob uploads, that don't have metadata id reserved. Before enabling "routerReservedMetadataEnabled" we need to get evidence that all cases are covered and there is no flow that inadvertently not reserved metadata chunk. That's why we need the metric even though "routerReservedMetadataEnabled" is disabled.